packages/compass-collection/src/components/collection-header/fle-badge.tsx

```diff
@@ -0,0 +1,23 @@
+import React from 'react';
+import {
+  Badge,
+  BadgeVariant,
+  css,
+  spacing,
+} from '@mongodb-js/compass-components';
+
+const collectionHeaderBadgeStyles = css({
+  marginLeft: spacing[2],
+});
+
+const FLEBadge = (): React.ReactElement => (
+  <Badge
+    data-testid="collection-badge-fle"
+    className={collectionHeaderBadgeStyles}
+    variant={BadgeVariant.DarkGray}
+  >
+    FLE2
```

Comment:
Maybe we can use the icon we also use for CSFLE connections in the sidebar and encrypted fields in the CRUD view here, to give the user a consistent visual hint that something is FLE-related? Some of the other badges also have icons, so I think that might fit nicely :)

```suggestion
    {/* TODO(COMPASS-5626): use proper name instead of FLE2 */}
    <Icon glyph="Key" title="FLE2" />
    &nbsp;FLE2
```

(the TODO comment is optional, I’ve left one in similar places as well so that it’s easier to look up what exactly we’ll need to change once the public name is final)